test/mocha/core/curlify.js

```diff
@@ -194,6 +194,54 @@ describe("curlify", function() {
         expect(curlified).toEqual("curl -X POST \"http://example.com\" -H  \"content-type: multipart/form-data\" -F \"id=123\" -F \"file=@file.txt;type=text/plain\"")
     })
 
+    it("should print a curl with formData and file, and trailing custom header", function () {
+        // use case: requestInterceptor appends req.headers as trailing header
+        var file = new win.File()
+        file.name = "file.txt"
+        file.type = "text/plain"
+
+        var req = {
+            url: "http://example.com",
+            method: "POST",
+            headers: {
+                "content-type": "multipart/form-data",
+                "x-custom-name": "any-value"
+            },
+            body: {
+                id: "123",
+                file
+            }
+        }
+
+        let curlified = curl(Im.fromJS(req))
+
+        expect(curlified).toEqual("curl -X POST \"http://example.com\" -H  \"content-type: multipart/form-data\" -H  \"x-custom-name: any-value\" -F \"id=123\" -F \"file=@file.txt;type=text/plain\"")
+    })
+
+    it("should print a curl with formData and file, and leading custom header", function () {
+        // confirm leading custom header order is also valid, and does not impact -F
```

Comment:
Shouldn't use comments, unless absolutely necessary. Everything should be translated into test descriptions.

If we're testing for another aspect of generated curt, we should have a separate test for it.